### ISSUE TITLE
noqa comment parsing does not work for multi-line YAML commands

### ISSUE THREAD

**Issue by nre-ableton** [ansible-community/ansible-lint#1547]
##### Summary

The following `noqa` suppression works perfectly with ansible-lint:

```yaml
- name: Install Virtualenv
  become: true
  command: pip-3.8 install --upgrade virtualenv # noqa no-changed-when
```

However, this one does not:

```yaml
- name: Install Virtualenv
  become: true
  command: >  # noqa no-changed-when
    pip-3.8 install --upgrade --prefix='/usr/local' virtualenv
```

In the latter case, the `no-changed-when` violation is raised.

I'm pretty sure this hasn't been reported yet, but https://github.com/ansible-community/ansible-lint/issues/1506 might be related.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console (paste below)
$ ansible --version
ansible 2.10.5
  config file = /home/nre/Code/<redacted>/ansible.cfg
  configured module search path = ['/home/nre/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nre/Code/caches/venv/<redacted>/lib/python3.8/site-packages/ansible
  executable location = /home/nre/Code/caches/venv/<redacted>/bin/ansible
  python version = 3.8.6 (default, Jan 27 2021, 15:42:20) [GCC 10.2.0]

$ ansible-lint --version
ansible-lint 5.0.7 using ansible 2.10.5


```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### OS / ENVIRONMENT

Ubuntu Linux 20.04.

**Comment by ssbarnea** [comment]
Closing as no fix due to technical complexity involved with it, with note that #1584 should provide a more flexible solution for the future.

If someone things he can fix it, I would be glad to review a PR for it.

**Comment by simontunnat** [comment]
I have a related problem.
In my case I was not able to use the "uri" module and had a very long command line which I had to split in multiple lines.

This was working on 5.3.2:
```
        - name: "Upload"
          ansible.builtin.command: # noqa command-instead-of-module
            cmd: >
              curl [VERY LONG LIST OF ARGUMENTS]
```

But with 5.4.0 I get the following error message:
```
WARNING  Listing 1 violation(s) that are fatal
command-instead-of-module: curl used in place of get_url or uri module
roles/XXX.yml:96 Task/Handler: Upload

You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - command-instead-of-module  # Using command rather than module
```

So something changed in 5.4 as it was working before.
Is this something you would fix?
